docs/configuration.md

```diff
@@ -0,0 +1,270 @@
+# Connector Configuration Reference
+
+The AlloyDB Java Connector internally manages one or more Connectors. Most
+applications don't need to configure these connectors. However, when an  
+application needs advanced configuration or lifecycle management of the 
+AlloyDB Connector, then the application may need to configure multiple 
+AlloyDB Java Connectors. 
+
+Each AlloyDB Java Connector is responsible for establishing connections to the
+application's AlloyDB instances. A Connector holds a distinct Google Cloud IAM 
```

Comment:
Credentials -- I think "IAM" isn't technically correct here. If the same phrase is used in our public docs, then this is OK. Otherwise let's just drop "IAM."

Reply:
updated